### PR TITLE
Remove unsafe access to .data() and _data in Matrix

### PR DESCRIPTION
**Pull request by jkflying**
This PR removes access to the unsafe .data() and _data in Matrix.

Linked PRs which should be merged first:

~~https://github.com/PX4/Matrix/pull/90~~ Merged
~~https://github.com/PX4/ecl/pull/641~~ Merged

### CODE REVIEW

**Comment by dagar** [comment]
Nice!

File: src/modules/mavlink/mavlink_messages.cpp [hunk 1]
```diff
@@ -3431,7 +3431,10 @@ class MavlinkStreamAttitudeTarget : public MavlinkStream
 
 			} else {
 				matrix::Quatf q = matrix::Eulerf(att_sp.roll_body, att_sp.pitch_body, att_sp.yaw_body);
-				memcpy(&msg.q[0], q.data(), sizeof(msg.q));
+
+				for (size_t i = 0; i < 4; i++) {
```
Comment (dagar) [inline src/modules/mavlink/mavlink_messages.cpp]:
```suggestion
				q.copyTo(msg.q);
```

**Comment by jkflying** [comment]
Now depends also on https://github.com/PX4/Matrix/pull/94

**Comment by MaEtUgR** [comment]
We have to update the ecl again after it get merged.

File: src/systemcmds/tests/test_matrix.cpp [hunk 2]
```diff
@@ -737,7 +739,7 @@ bool MatrixTest::dcmRenormTests()
 
 	if (verbose) {
 		for (int row = 0; row < 3; row++) {
-			matrix::Vector3f rvec(A._data[row]);
+			matrix::Vector3f rvec(matrix::Matrix<float, 1, 3>(A.row(row)).transpose());
```
Comment (MaEtUgR) [inline src/systemcmds/tests/test_matrix.cpp]: No need for `matrix::` there's `using namespace` at the top.

File: src/systemcmds/tests/test_matrix.cpp [hunk 3]
```diff
@@ -749,7 +751,7 @@ bool MatrixTest::dcmRenormTests()
 	err = 0.0f;
 
 	for (int row = 0; row < 3; row++) {
-		matrix::Vector3f rvec(A._data[row]);
+		matrix::Vector3f rvec(matrix::Matrix<float, 1, 3>(A.row(row)).transpose());
```
Comment (MaEtUgR) [inline src/systemcmds/tests/test_matrix.cpp]: same here

**Pull request merged**